Tella/Scenes/Uwazi/ViewModel/UwaziEntityViewModel.swift

```diff
@@ -34,5 +40,59 @@ class UwaziEntityViewModel: ObservableObject {
         requiredPrompts.forEach { prompt in
             prompt.showMandatoryError = prompt.value.stringValue.isEmpty
         }
+        
+        if(!requiredPrompts.isEmpty) {
+            submitEntity()
+        }
+    }
+    
+    private func submitEntity() {
+        // Extract entity data and metadata
+        let entityData = extractEntityDataAndMetadata()
+        
+        // Prepare server URL and cookie list
+        let serverURL = self.serverURL
+        let cookieList = ["connect.sid=" + self.accessToken]
+        
+        // Submit the entity data
+        let response = UwaziServerRepository().submitEntity(serverURL: serverURL, cookieList: cookieList, entity: entityData)
+               response.sink { completion in
+                   switch completion {
+
+                   case .finished:
+                       print("Finished")
+                   case .failure(let error):
+                       print(error)
+                   }
+                   } receiveValue: { value in
+                       print(value)
+                   }
+
+                   .store(in: &subscribers)
+    }
+
+    private func extractEntityDataAndMetadata() -> ([String: Any]) {
+        var entityData: [String: Any] = [:]
+        var metadata: [String: Any] = [:]
+
+        for entryPrompt in entryPrompts {
+            switch UwaziEntityPropertyType(rawValue: entryPrompt.type) {
+            case .dataTypeText:
+                if entryPrompt.name == "title" {
+                    entityData[entryPrompt.name!] = entryPrompt.value.stringValue
```

Comment:
Try to avoid the forced unwrapping, because in case the value is nil the app will crash

Reply:
Thanks for the feedback. I fixed this on the file selector [PR](https://github.com/Horizontal-org/Tella-iOS/pull/105)